src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Animation/Generated/TimelineCollection.cs

```diff
@@ -926,63 +926,58 @@ public TimelineCollection(IEnumerable<Timeline> collection)
 
             WritePreamble();
 
-            if (collection != null)
+            ArgumentNullException.ThrowIfNull(collection);
+
+            bool needsItemValidation = true;
+            ICollection<Timeline> icollectionOfT = collection as ICollection<Timeline>;
+
+            if (icollectionOfT != null)
+            {
+                _collection = new FrugalStructList<Timeline>(icollectionOfT);
+            }
+            else
             {
-                bool needsItemValidation = true;
-                ICollection<Timeline> icollectionOfT = collection as ICollection<Timeline>;
+                ICollection icollection = collection as ICollection;
 
-                if (icollectionOfT != null)
+                if (icollection != null) // an IC but not and IC<T>
                 {
-                    _collection = new FrugalStructList<Timeline>(icollectionOfT);
+                    _collection = new FrugalStructList<Timeline>(icollection);
                 }
-                else
-                {       
-                    ICollection icollection = collection as ICollection;
+                else // not a IC or IC<T> so fall back to the slower Add
+                {
+                    _collection = new FrugalStructList<Timeline>();
 
-                    if (icollection != null) // an IC but not and IC<T>
-                    {
-                        _collection = new FrugalStructList<Timeline>(icollection);
-                    }
-                    else // not a IC or IC<T> so fall back to the slower Add
+                    foreach (Timeline item in collection)
                     {
-                        _collection = new FrugalStructList<Timeline>();
-
-                        foreach (Timeline item in collection)
+                        if (item == null)
                         {
-                            if (item == null)
-                            {
-                                throw new System.ArgumentException(SR.Collection_NoNull);
-                            }
-                            Timeline newValue = item;
-                            OnFreezablePropertyChanged(/* oldValue = */ null, newValue);
-                            _collection.Add(newValue);
-
+                            throw new System.ArgumentException(SR.Collection_NoNull);
```

Comment:
why can't we use ArgumentException.ThrowIfNull() here?

Reply:
same reasoning as above